File: pierre_feuille_ciseaux/python/pfc.py
import random

# ordinateur joue aléatoirement
def randomPlayer():
    print("donner le chiffre entre 1 et 3 1:pierre 2: feuille 3:ciseau")
    return random.randint(1,3)

# jeu de l'humain
def humanPlayer():
    print("donner le chiffre entre 1 et 3 1:pierre 2: feuille 3:ciseau")
    n = input()
    if (n  != '1') and (n != '2') and (n != '3'):
        humanPlayer()
    return int(n)

# on choisit si c'est un humain ou l'ordinateur
def choixjoueur(joueur):
    
    print(f"{joueur} : ordinateur ou humain (o/h)")
    n = input()
    
    # si n est different de o et h
    if (n != 'o') and (n != 'h'):
        n = choixjoueur(joueur)
        
    match(n):
    
        case 'o':
            player = randomPlayer
        case 'h':
            player = humanPlayer
        
    return player

# on choisit les 2 joueurs
def choix_joueur():
    player1 = choixjoueur("premier joueur")
    player2 = choixjoueur("deuxieme joueur")
    return player1,player2
    

# pierre-feuille-ciseau
def pfc(maxPoint):
    player1,player2 = choix_joueur()
    
    pointPlayer1 = 0
    pointPlayer2 = 0

    liste = ("la pierre", "la feuille", "le ciseau")
    while pointPlayer1 < maxPoint and pointPlayer2 < maxPoint:
        cpt1 = player1()
        print(f" joueur 1 a joué {liste[cpt1-1]}\n")
        cpt2 = player2()
        print(f" joueur 2 a joué {liste[cpt2-1]}\n")
        
        cpt = cpt1 - cpt2
        
        match(cpt):
            case -2:
                print(f"{liste[cpt1-1]} a gagné contre {liste[cpt2-1]}\n")
                pointPlayer1 += 1
            case -1:
                print(f"{liste[cpt1-1]} a perdu contre {liste[cpt2-1]}\n")
                pointPlayer2 += 1
            case 0:
                print(f"les 2 joueurs ont joués {liste[cpt1-1]}\n")
                print(f"player1 : { pointPlayer1}  player2 : { pointPlayer2}\n")
                continue
            case 1:
                print(f"{liste[cpt1-1]} a gagné contre {liste[cpt2-1]}\n")
                pointPlayer1 += 1
            case 2:
                print(f"{liste[cpt1-1]} a perdu contre {liste[cpt2-1]}\n")
                pointPlayer2 += 1
        
        
        print(f"player1 : { pointPlayer1}  player2 : { pointPlayer2}\n")
    if pointPlayer1 > pointPlayer2:
        print("joueur 1 a gagné")

    else:
         print("joueur 2 a gagné")

File: pierre_feuille_ciseaux/C/pfc.c
#include "pfc.h"

/* ordinateur joue aléatoirement*/
int randomPlayer(){
    
    srand(time(NULL));
    
    int i;
    
    printf("donner le chiffre entre 1 et 3 1:pierre 2: feuille 3:ciseau\n");
    
    /* chiffre entre 1 et 3 pris aléatoirement*/
    
    i = rand()%3+1;
    printf("%d\n",i);
    return i;
}

/* jeu avec l'humain*/
int humanPlayer(){
    int choix;
    printf("donner le chiffre entre 1 et 3 1:pierre 2: feuille 3:ciseau\n");
    scanf("%d",&choix);
    
    
    
    if ((choix  != 1) && (choix != 2) && (choix != 3)){
        humanPlayer();
    }
    
    return choix;
}


/* on choisit si c'est un humain ou l'ordinateur */
char choixjoueur(char nomplayer[10]){
    char choix;
    printf("%s joueur: ordinateur ou humain (o/h)\n",nomplayer);
    scanf("%c", &choix);
    if ((choix != 'o') && (choix != 'h')){
        choix = choixjoueur(nomplayer);
   }
    return choix;
    
}

/* choisir le joueur soit humain ou random*/
int choix_joueur(char player){
    int jeuPlayer = 0;
    switch(player){
        case 'o':
            jeuPlayer = randomPlayer();
            break;
        case 'h':
            jeuPlayer = humanPlayer();
            break;
    }
    return jeuPlayer;
    
}
/*voir le probleme sur humanPlayer et sur switch*/
/* Pierre Feuille Ciseau*/
void pfc(int maxPoint){
    char player1, player2;
    
    player1 = choixjoueur("Premier");
    player2 = choixjoueur("Deuxieme");
    
    
    int pointPlayer1 = 0;
    int pointPlayer2 = 0;
    
    char * liste[3] = {"la Pierre", "la Feuille", "le Ciseau"};
    
    int jeuPlayer1, jeuPlayer2, differencePlayer;
   
    
    while ((pointPlayer1 < maxPoint) && (pointPlayer2 < maxPoint)){
        jeuPlayer1 = choix_joueur(player1);
        printf("Le joueur 1 a joué %s\n", liste[jeuPlayer1 -1]);
        
        jeuPlayer2 = choix_joueur(player2);
        printf("Le joueur 2 a joué %s\n", liste[jeuPlayer2 -1]);
        
        
        differencePlayer = jeuPlayer1 - jeuPlayer2;
       
        switch(differencePlayer){
            case -2 :
                
                printf("%s a gagné contre %s\n", liste[jeuPlayer1 -1], liste[jeuPlayer2 -1]);
                pointPlayer1 += 1;
                break;
                
            case -1:
                
                printf("%s a perdu contre %s\n", liste[jeuPlayer1 -1], liste[jeuPlayer2 -1]);
                pointPlayer2 += 1;
                break;
                
            case 0:
               
                printf("Les 2 joueurs ont joués %s\n", liste[jeuPlayer1 -1]);
                printf("Player1 : %d player2 : %d\n", pointPlayer1, pointPlayer2);
                break;
            
            case 1:
                
                printf("%s a gagné contre %s\n", liste[jeuPlayer1 -1], liste[jeuPlayer2 -1]);
                pointPlayer1 += 1;
                break;
            
            case 2:
                
                printf("%s a perdu contre %s\n", liste[jeuPlayer1 -1], liste[jeuPlayer2 -1]);
                pointPlayer2 += 1;
                break;
        }
        
        printf("Player1 : %d player2 : %d\n", pointPlayer1, pointPlayer2);
        
    }
           if (pointPlayer1 > pointPlayer2){
        printf("joueur 1 a gagné\n");
    }
           else{
        printf("joueur 2 a gagné\n");
    }
}

